modules/base_test.py

```diff
@@ -0,0 +1,69 @@
+"""
+Base class definitions to enable consistent logging. Requires ReFrame v. 3.11.x
+"""
+import math
+import reframe as rfm
+from reframe.core.decorators import run_after
+
+
+class DiRACTest(rfm.RegressionTest):
+
+    num_total_cores = variable(int, loggable=True)
+    num_omp_threads = variable(int, loggable=True)
+    num_mpi_tasks = variable(int, loggable=True)
+    num_mpi_tasks_per_node = variable(int, loggable=True)
+    num_nodes = variable(int, loggable=True)
+
+    @run_after('setup')
+    def set_attributes_after_setup(self):
+        """Set the required MPI and OMP ranks/tasks/threads"""
+
+        self.num_mpi_tasks = self.num_tasks = max(self.num_total_cores//self.num_omp_threads, 1)
+
+        try:
+            num_cores_per_node = self._current_partition.processor.num_cpus
```

Comment:
`self._current_partition` looks an internal attribute?  The user-facing one is [`self.current_partition`](https://reframe-hpc.readthedocs.io/en/stable/regression_test_api.html?#reframe.core.pipeline.RegressionTest.current_partition).  I don't think accessing this should ever result in a `AttributeError`.

Reply:
This is an excellent point – I'm sure there was a reason for accessing the private attr but it escapes me now. If it works with the public one then feel free to change

Something in that try definitely did, but this was long ago and probably with a different ReFrame version so 🤷